src/sentry/sentry_apps/tasks/sentry_apps.py

```diff
@@ -48,64 +253,248 @@ def send_alert_event(
+) -> None:
+    webhook_data = get_webhook_data(installation_id, issue_id, user_id)
+    if not webhook_data:
+        return
+    install, issue, user = webhook_data
+    data = kwargs.get("data", {})
+    data.update({"issue": serialize(issue)})
+    send_webhooks(installation=install, event=f"issue.{type}", data=data, actor=user)
+    analytics.record(
+        f"sentry_app.issue.{type}",
         user_id=user_id,
-        *args,
-        **kwargs,
+        group_id=issue_id,
+        installation_id=installation_id,
     )
 
 
 @instrumented_task(
     name="sentry.sentry_apps.tasks.sentry_apps.build_comment_webhook", **TASK_OPTIONS
 )
 @retry_decorator
-def build_comment_webhook(installation_id, issue_id, type, user_id, *args, **kwargs):
-    old_build_comment_webhook(
-        installation_id=installation_id,
-        issue_id=issue_id,
-        type=type,
+def build_comment_webhook(
+    installation_id: int, issue_id: int, type: str, user_id: int, *args: Any, **kwargs: Any
+) -> None:
+    webhook_data = get_webhook_data(installation_id, issue_id, user_id)
+    if not webhook_data:
+        return None
+    install, _, user = webhook_data
+    data = kwargs.get("data", {})
+    project_slug = data.get("project_slug")
+    comment_id = data.get("comment_id")
+    payload = {
+        "comment_id": data.get("comment_id"),
+        "issue_id": issue_id,
+        "project_slug": data.get("project_slug"),
+        "timestamp": data.get("timestamp"),
+        "comment": data.get("comment"),
+    }
+    send_webhooks(installation=install, event=type, data=payload, actor=user)
+    # `type` is comment.created, comment.updated, or comment.deleted
+    analytics.record(
+        type,
         user_id=user_id,
-        *args,
-        **kwargs,
+        group_id=issue_id,
+        project_slug=project_slug,
+        installation_id=installation_id,
+        comment_id=comment_id,
     )
 
 
+def get_webhook_data(
+    installation_id: int, issue_id: int, user_id: int
+) -> tuple[RpcSentryAppInstallation, Group, RpcUser | None] | None:
+    extra = {"installation_id": installation_id, "issue_id": issue_id}
+    install = app_service.installation_by_id(id=installation_id)
+    if not install:
+        logger.info("workflow_notification.missing_installation", extra=extra)
+        return None
+
+    try:
+        issue = Group.objects.get(id=issue_id)
+    except Group.DoesNotExist:
+        logger.info("workflow_notification.missing_issue", extra=extra)
+        return None
+
+    user = None
+    if user_id:
+        user = user_service.get_user(user_id=user_id)
+        if not user:
+            logger.info("workflow_notification.missing_user", extra=extra)
+
+    return (install, issue, user)
+
+
 @instrumented_task(
-    "sentry.sentry_apps.tasks.sentry_apps.send_resource_change_webhook", **TASK_OPTIONS
+    "sentry.sentry_apps.tasks.sentry_apps.send_process_resource_change_webhook", **TASK_OPTIONS
```

Comment:
Renaming this task could result in dropped tasks as a worker with the new code could create a task that is received by a worker with the old code where this task name doesn't exist.

Reply:
F, thank you for catching this.